File: src/go/partition/l/seq.go
package large

func large_sum_seq(arr []int, size int, index int64) int {
	var sum int = 0

	for i := 0; i < size; i++ {
		if index&(1<<i) != 0 {
			sum += arr[i]
		}
	}

	return sum
}

func Seq(arr []int) int32 {
	var combs int64 = 1 << (len(arr) - 1)
	var all int64 = (1 << len(arr)) - 1

	var problem_sum int = large_sum_seq(arr, len(arr), all)
	if problem_sum%2 != 0 {
		return 0
	}
	var half_sum = problem_sum / 2

	for i := int64(1); i < combs; i++ {
		var sum int = large_sum_seq(arr, len(arr), i)
		if sum == half_sum {
			return 1
		}
	}

	return 0
}

File: src/julia/partition/l/seq.jl
module Seq

export large_seq

function large_sum_seq(arr::Vector{Int64}, size::Int, index::Int64) 
    sum = 0;

    for i in 1:size
        if index & (1 << (i - 1)) != 0
            sum += arr[i]
        end
    end

    return sum
end # large_sum_seq

function large_seq(arr::Vector{Int64})
    size = length(arr)
    combs = 1 << (size - 1)
    all = (1 << size) - 1

    problem_sum = large_sum_seq(arr, size, all)
    if problem_sum % 2 != 0
        return [false]
    end
    half_sum = problem_sum / 2

    for i in 1:combs
        sum = large_sum_seq(arr, size, i)
        if sum == half_sum
            return [true]
        end
    end

    return [false]
end


end # module